src/map/mob.cpp

```diff
@@ -5463,59 +5451,115 @@ uint64 MobAvailDatabase::parseBodyNode(const YAML::Node &node) {
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
+ */
+uint64 MobSummonDatabase::parseBodyNode(const YAML::Node &node) {
+	if (!this->nodesExist(node, { "Group" })) {
+		return 0;
 	}
+	std::string group_name;
 
-	mob_id = atoi(str[1]);
-	if (mob_id != 0 && mob_db.find(mob_id) == nullptr) {
-		ShowError("mob_readdb_group: Invalid random monster group '%s'\n", str[0]);
-		return false;
+	if (!this->asString(node, "Group", group_name))
+		return 0;
+
+	std::string group_name_constant = "MOBG_" + group_name;
+	int64 constant;
+
+	if (!script_get_constant(group_name_constant.c_str(), &constant) || constant < MOBG_Branch_Of_Dead_Tree || constant > MOBG_Taekwon_Mission) {
+		this->invalidWarning(node["Group"], "Invalid monster group %s.\n", group_name.c_str());
+		return 0;
 	}
-	else if (mob_id == 0){
-		mob_id = atoi(str[3]);
-		if (mob_db.find(mob_id) == nullptr) {
-			ShowError("mob_readdb_group: Invalid random monster group '%s'\n", str[0]);
-			return false;
-		}
-		set_default = true;
+
+	uint8 id = static_cast<uint8>(constant);
+
+	std::shared_ptr<s_randomsummon_group> summon = this->find(id);
+	bool exists = summon != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "Default" }))
+			return 0;
+
+		summon = std::make_shared<s_randomsummon_group>();
+		summon->random_id = id;
 	}
 
-	if (!(msummon = (struct s_randomsummon_group *)idb_get(mob_summon_db, group))) {
-		CREATE(msummon, struct s_randomsummon_group, 1);
-		CREATE(msummon->list, struct s_randomsummon_entry, (msummon->count = 1));
-		msummon->list[0].mob_id = mob_id;
-		msummon->list[0].rate = atoi(str[3]);
-		msummon->random_id = group;
-		idb_put(mob_summon_db, group, msummon);
+	if (this->nodeExists(node, "Default")) {
+		std::string mob_name;
+
+		if (!this->asString(node, "Default", mob_name))
+			return 0;
+
+		std::shared_ptr<s_mob_db> mob = mobdb_search_aegisname(mob_name.c_str());
+
+		if (mob == nullptr) {
+			this->invalidWarning(node["Default"], "Unknown mob %s.\n", mob_name.c_str());
+			return 0;
+		}
+		summon->default_mob_id = mob->vd.class_;
 	}
-	else {
-		ARR_FIND(0, msummon->count, i, set_default || (i > 0 && msummon->list[i].mob_id == mob_id));
-		if (i >= msummon->count)
-			RECREATE(msummon->list, struct s_randomsummon_entry, ++msummon->count);
-		msummon->list[i].mob_id = mob_id;
-		msummon->list[i].rate = atoi(str[3]);
+
+	if (this->nodeExists(node, "Summon")) {
+		const YAML::Node &MobNode = node["Summon"];
+
+		for (const YAML::Node &mobit : MobNode) {
+			if (!this->nodesExist(mobit, { "Mob", "Rate" })) {
+				continue;
+			}
+
+			std::string mob_name;
+
+			if (!this->asString(mobit, "Mob", mob_name))
+				continue;
+
+			std::shared_ptr<s_mob_db> mob = mobdb_search_aegisname(mob_name.c_str());
+
+			if (mob == nullptr) {
+				this->invalidWarning(mobit["Mob"], "Unknown mob %s.\n", mob_name.c_str());
+				continue;
+			}
+
+			uint32 rate;
+
+			if (!this->asUInt32(mobit, "Rate", rate))
```

Comment:
I think it might be good to change the way we store "Rate" here.
Do we really need it to be that detailed?
I think our UInt16Rate would be sufficient. But you might have to use UInt16 for the "remove" feature.

Reply:
The tables are using the values from aegis, for example
```
      - Mob: HORN
        Rate: 357142
```
need int32 or I'm missing something?

Reply:
Haha I did not check the entries. Guess we have to stick with stupid AEGIS like logic then.